gnd/src/main/java/com/google/android/gnd/workers/FileDownloadWorkManager.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.workers;
+
+import androidx.lifecycle.LiveData;
+import androidx.work.Constraints;
+import androidx.work.ExistingWorkPolicy;
+import androidx.work.NetworkType;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkInfo;
+import androidx.work.WorkManager;
+
+/** Enqueues file download work to be done in the background. */
+public class FileDownloadWorkManager {
+  private final WorkManager workManager;
+
+  public FileDownloadWorkManager() {
```

Comment:
Hmm I was trying that before, to no avail. Then again, I was calling it directly in the MapsAdapter--since it was a faster way to demo and check that it functioned as expected. Perhaps that was messing it up?

Reply:
Right. `GoogleMapsMapAdapter` itself isn't in the Dagger graph (ie it's not injected), so injecting it there wouldn't work.